### PR TITLE
Add firewall policy premium feature properties

### PR DESCRIPTION
**Pull request by adgrieve**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/firewallPolicy.json [hunk 1]
```diff
@@ -1114,6 +1137,170 @@
...
+        "modelAsString": true
+      }
+    },
+    "FirewallPolicyTransportSecurity": {
+      "properties": {
+        "certificateAuthority": {
+          "$ref": "#/definitions/FirewallPolicyCertificateAuthority",
+          "description": "The CA used for intermediate CA generation."
+        },
+        "excludedDomains": {
```
Comment (gersner) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/firewallPolicy.json:1272]: @adgrieve please remove, we moved Cx exclusion to P1

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/firewallPolicy.json [hunk 2]
```diff
@@ -1114,6 +1137,170 @@
...
+          "type": "string",
+          "description": "Name of the bypass traffic rule."
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the bypass traffic rule."
+        },
+        "protocol": {
+          "type": "string",
+          "$ref": "#/definitions/FirewallPolicyIntrusionSystemBypassTrafficProtocol",
```
Comment (gersner) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/firewallPolicy.json:1211]: Should this be FirewallPolicyIntrusionSystem**Detection**BypassTrafficProtocol?

**Comment by lirenhe** [comment]
@adgrieve, could you take a look at the model validation error? Seems in the example, there is an additional property of keyVaultSecretId.

**Comment by adgrieve** [comment]
@MikhailTryakhov what is the expected release date of this branch?

**Comment by lirenhe** [comment]
@adgrieve, the PR looks good to me, is it ready to merge?

**Review by lirenhe — APPROVED** [review]

**Comment by adgrieve** [comment]
> @adgrieve, the PR looks good to me, is it ready to merge?

@lirenhe I can't find exact release dates for this PR anywhere. Do you happen to have the time table for that?

**Comment by lirenhe** [comment]
> > @adgrieve, the PR looks good to me, is it ready to merge?
> 
> @lirenhe I can't find exact release dates for this branch anywhere. Do you happen to have the time table for that?

I have no idea of that.  i would only merge it to Azure:network-july-release

**Pull request merged**